src/PalindromeNumber/Solution.cpp

```diff
@@ -0,0 +1,78 @@
+// Solution by Imtiaz Ahmed (Github: tiazahmd)
+
+#include <iostream>
+#include <vector>
+
+#include "../Test.h"
+
+using namespace std;
+using namespace leetcode;
+
+class Solution {
+public:
+  string isPalindrome(int x) {
+    if (x >= 0 && x < 10) {
+      return "true";
+    } else if (x >= 10) {
+      vector<int> vec;
+      long div = 10;
+      long ct = 1;
+      int num = 0;
+      int digCount = 0;
+      int temp = x;
+
+      while (temp != 0) {
+        temp /= 10;
+        digCount++;
+      }
+
+      for (int i = 0; i < digCount; i++) {
+        num = x % div;
+        num /= ct;
+        div *= 10;
+        ct *= 10;
+        vec.push_back(num);
+      }
+
+      for (int i = 0; i < vec.size(); i++) {
```

Comment:
according your algorithm, `i` reaches the middle of the vector.

Reply:
`i` is not supposed to go beyond half way. It compares the `first` and `last` integer, then the `(first + 1)`th and `(last - 1)`th digit and so on. In order to do that, the loop does not need to go beyond half way through.

Reply:
for  example, vec ＝ { 1, 2, 3, 2, 1}, i will to go 4.

Reply:
So far, what I can see is that it is working as a "failsafe" so that if there are even number of integers, it will make sure that integer is accounted for. I can't seem to find a more elegant way to do this without compromising speed.

Reply:
@Zhongnibug What do you think？